### PR TITLE
CT 1537 fix event test and rename a couple of fields

### LINKED ISSUE #6292: [CT-1537] Fix logging event test and rename a few fields
While working on the pull request for #6256, issues were fixed in the logging event tests and a few issues with event names were fixed. This is a ticket to process that work.

### PR DESCRIPTION
**Pull request by gshank**
resolves #6292

### Description

The event unit test was broken and some of the tests weren't actually doing anything. This fixes that, removes the lists of events from the bottom of the types.py and test_types.py files and moves them into test_events.py test file. It also renames a few tests.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [x] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/context/base.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
     disallow_secret_env_var,
 )
 from dbt.events.functions import fire_event, get_invocation_id
-from dbt.events.types import MacroEventInfo, MacroEventDebug
+from dbt.events.types import JinjaLogInfo, JinjaLogDebug
```
Comment (emmyoop) [inline core/dbt/context/base.py:19]: Along these same lines there's also a `GeneralMacroWarning` that is the catchall for `warn` defined in `exceptions.py`.  It's worth renaming that along with these.

File: core/dbt/events/types.proto [hunk 2]
```diff
@@ -15,6 +15,7 @@ message EventInfo {
     string thread = 7;
     google.protobuf.Timestamp ts = 8;
     map<string, string> extra = 9;
+    string category = 10;
```
Comment (emmyoop) [inline core/dbt/events/types.proto:18]: Is this eventually going to be the log category?  Am I missing how this gets populated?

**Review by emmyoop — CHANGES_REQUESTED** [review]
One more rename and one question.

File: core/dbt/events/types.proto (hunk 2 above)
Comment (gshank) [reply core/dbt/events/types.proto:18]: Yeah, right now it's just a placeholder. Eventually we'll populate it somehow...

File: core/dbt/context/base.py (hunk 1 above)
Comment (gshank) [reply core/dbt/context/base.py:19]: It doesn't look like it's actually used anywhere. Can we just remove it?

File: core/dbt/events/types.proto (hunk 2 above)
Comment (emmyoop) [reply core/dbt/events/types.proto:18]: Is there an issue so we don't lose the fact we still need to actually populate it?

File: core/dbt/context/base.py (hunk 1 above)
Comment (emmyoop) [reply core/dbt/context/base.py:19]: It's used within macros ([docs](https://docs.getdbt.com/reference/dbt-jinja-functions/exceptions#warn))

Comment (gshank) [reply core/dbt/context/base.py:19]: So it's exported into the context. So should we call in JinjaLogWarning?

Comment (gshank) [reply core/dbt/context/base.py:19]: Or JinjaLogWarn?

Comment (gshank) [reply core/dbt/context/base.py:19]: I went with JinjaLogWarning.

File: core/dbt/events/types.proto (hunk 2 above)
Comment (gshank) [reply core/dbt/events/types.proto:18]: Mostly it's your ticket #5958.

**Pull request merged**